src/Arcus.EventGrid.Publishing/EventGridPublisher.cs

```diff
@@ -256,14 +297,26 @@ public async Task PublishManyAsync<TEvent>(IEnumerable<TEvent> events)
             return content;
         }
 
-        private async Task PublishContentToTopicAsync(HttpContent content)
+        private async Task PublishContentToTopicAsync(HttpContent content, string subject)
         {
-            using (HttpResponseMessage response = 
-                await _resilientPolicy.ExecuteAsync(() => SendAuthorizedHttpPostRequestAsync(content)))
+            using (DependencyMeasurement measurement = DependencyMeasurement.Start())
             {
-                if (!response.IsSuccessStatusCode)
+                var isSuccessful = false;
+                try
+                {
+                    using (HttpResponseMessage response =
+                        await _resilientPolicy.ExecuteAsync(() => SendAuthorizedHttpPostRequestAsync(content)))
+                    {
+                        isSuccessful = response.IsSuccessStatusCode;
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            await ThrowApplicationExceptionAsync(response);
+                        }
+                    }
+                }
+                finally
                 {
-                    await ThrowApplicationExceptionAsync(response);
+                    _logger.LogDependency("Azure Event Grid", dependencyData: subject ?? "<no-subject>", targetName: TopicEndpoint, isSuccessful: isSuccessful, measurement);
```

Comment:
I'm not sure if the subject or the event type should be the dependency data 🤔

Reply:
I'm no expert, but is the dependency data something more info about 'identifying' the event? We can always add the event type as a property in the context, right?

Reply:
It helps better understand what was being used for that dependency, more conceptual info